protoc-gen-eventuals/protoc-gen-eventuals.py

```diff
@@ -34,6 +34,22 @@ def analyze_file(self, proto_file: FileDescriptorProto) -> dict:
                 for service in proto_file.service
             ]
         }
+        
+        header_type = {}
+        for service in file_digest['services']:
+            for method in service['methods']:
+              for client_streaming in [True, False]:
+                for server_streaming in [True, False]:
+                  if method['server_streaming'] == server_streaming and method['client_streaming'] == client_streaming:
+                    header_type[(client_streaming, server_streaming)] = True
+
+        header_type['has_bidirectional_streaming'] = header_type.pop((True, True), False)
+        header_type['has_server_streaming'] = header_type.pop((False, True), False)
+        header_type['has_client_streaming'] = header_type.pop((True, False), False)
+       	header_type['has_no_streaming'] = header_type.pop((False, False), False) 
```

Comment:
I know I suggested these names, but I only now realize that they somewhat clash with the `method['server_streaming']` and `method['client_streaming']`...

Can we rename these variables (here and in the template) to the slightly longer but also more descriptive `bidirectional_streaming_methods_exists` (for the bidirectional case)?
I think this makes it clearer that it is not about the particular method but about the general existence of a particular type of methods.

To be clear, the naming of the templates I think is fine, it is just the variable name that could be a little crisper.